### ISSUE TITLE
New "Analyze Scope" action (use IDEA's AnalysisActionDialog and AnalysisScope)

### ISSUE THREAD

**Issue by GoogleCodeExporter** [andrepdo/findbugs-idea#47]
```
Basically the existing analyze actions scan/filter for .class files in the 
compiler output path (and then add to the FindBugs configuration).
This leads to a limitation in using same package names in different 
modules/projects, eg:

moduleA/src/com/company/util
moduleB/src/com/company/util

There is no way to just analzye "moduleB/src/com/company/util":
For example the "Analyze Package" action always scan both modules because all 
.class files are located at the same compiler output path.
(This is not bug, it does exactly what it should do: analyze the package)

The new action "Analyze Scope" should use IDEA's default analysis scope dialog:
com.intellij.analysis.BaseAnalysisActionDialog  (eg: IDEA->Analyze->Analyze 
Dependencies..)

In short:
The dialog return a com.intellij.analysis.AnalysisScope which supports the 
visitor pattern.
-> For every visited source file calculate the .class file path in the compiler 
output path and add it to the FindBugs configuration.
```

Original issue reported on code.google.com by `reto.merz@gmail.com` on 2 Dec 2013 at 10:51
